pkg/workspace/project.go

```diff
@@ -4,24 +4,44 @@
 package workspace
 
 import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
 	"fmt"
 	"strings"
 
 	"github.com/daytonaio/daytona/internal"
 	"github.com/daytonaio/daytona/pkg/gitprovider"
 )
 
+type ProjectBuildDevcontainer struct {
+	DevContainerFilePath string `json:"devContainerFilePath"`
+} // @name ProjectBuildDevcontainer
+
+type ProjectBuildDockerfile struct {
+	Context    string            `json:"context"`
+	Dockerfile string            `json:"dockerfile"`
+	Args       map[string]string `json:"args"`
+} // @name ProjectBuildDockerfile
+
+type ProjectBuild struct {
+	Devcontainer *ProjectBuildDevcontainer `json:"devcontainer"`
+	Dockerfile   *ProjectBuildDockerfile   `json:"dockerfile"`
+} // @name ProjectBuild
+
 type Project struct {
-	Name              string                     `json:"name"`
-	Image             string                     `json:"image"`
-	User              string                     `json:"user"`
-	Repository        *gitprovider.GitRepository `json:"repository"`
-	WorkspaceId       string                     `json:"workspaceId"`
-	ApiKey            string                     `json:"-"`
-	Target            string                     `json:"target"`
-	EnvVars           map[string]string          `json:"-"`
-	State             *ProjectState              `json:"state,omitempty"`
-	PostStartCommands []string                   `json:"postStartCommands,omitempty"`
+	Name               string                     `json:"name"`
+	Image              string                     `json:"image"`
+	User               string                     `json:"user"`
+	Build              *ProjectBuild              `json:"build"`
+	Repository         *gitprovider.GitRepository `json:"repository"`
+	WorkspaceId        string                     `json:"workspaceId"`
+	ApiKey             string                     `json:"-"`
+	Target             string                     `json:"target"`
+	EnvVars            map[string]string          `json:"-"`
+	State              *ProjectState              `json:"state,omitempty"`
+	PostCreateCommands []string                   `json:"postCreateCommands,omitempty"`
```

Comment:
Correct me if I'm wrong, but I don't see the `PostCreateCommands` being run in the agent.

Reply:
It needs to be implemented in the agent. Help appreciated :)

Reply:
Aah, okay. No problem, I'll add it asap.